dataInterface/movies.js

```diff
@@ -89,10 +80,67 @@ module.exports.create = async (newObj) => {
   }
 }
 
+// https://www.mongodb.com/docs/drivers/node/current/fundamentals/crud/write-operations/change-a-document/
+module.exports.updateById = async (movieId, newObj) => {
+  const database = client.db(databaseName);
+  const movies = database.collection(movieColl);
+
+  // Product team says only these two fields can be updated.
+  const updateRules = {
+    $set: {"title" : newObj.title, "plot": newObj.plot}
+  };
+  const filter = { _id: ObjectId(movieId) };
+  const result = await movies.updateOne(filter, updateRules);
+
+  if(result.modifiedCount != 1){
+    return {error: `Something went wrong. ${result.modifiedCount} movies were updated. Please try again.`}
+  };
+
+  const updatedMovie = module.exports.getById(movieId);
+  return updatedMovie;
+}
+
+// https://www.mongodb.com/docs/drivers/node/current/fundamentals/crud/write-operations/delete/
+module.exports.deleteById = async (movieId) => {
+  const database = client.db(databaseName);
+  const movies = database.collection(movieColl);
+
+  const deletionRules = {_id:ObjectId(movieId)}
+  const result = await movies.deleteOne(deletionRules);
+
+  if(result.deletedCount != 1){
+    return {error: `Something went wrong. ${result.deletedCount} movies were deleted. Please try again.`}
+  };
+
+  return {message: `Deleted ${result.deletedCount} movie.`};
+}
+
+// -------------------------- //
+// --- comments interface --- //
+// -------------------------- //
+
+module.exports.getAllComments = async (movieId) => {
+  const database = client.db(databaseName);
+  const comments = database.collection(commentColl);
+
+  const query = {movie_id: ObjectId(movieId)}
+  let commentCursor = await comments.find(query);
+  return commentCursor.toArray();
+}
+
+module.exports.getOneComment = async (commentId) => {
+  const database = client.db(databaseName);
+  const comments = database.collection(commentColl);
+
+  const query = {_id: ObjectId(commentId)}
+  let commentCursor = await comments.find(query);
+  return commentCursor.toArray();
```

Comment:
Alternatively, use the `findOne` method to get just the first match for this id, since we expect there to only be a single match.